e2e-robot/Makefile

```diff
@@ -0,0 +1,43 @@
+
+.PHONY: setup dry-run with-filter with-tag all
+
+setup:
+	python3 -m venv .venv && \
+	source .venv/bin/activate && \
+	python3 -m pip install -r requirements.txt
+
+dry-run:
+	source .venv/bin/activate && \
+	robot \
+		--dryrun \
+		--outputdir ../_output/e2e-$$(date +'%Y%m%d-%H%M%S') \
+		./tests/microshift.robot
+
+with-filter:
+	source .venv/bin/activate && \
+	robot \
+		-v USHIFT_IP:192.168.122.166 \
```

Comment:
The IP could come from running `./scripts/devenv-builder/manage-vm.sh ip`

Reply:
Yeah, it could also be passed via ENV vars or other means. I just hardcoded it to reduce typing when PoCing this :)
Not tying it to other scripts can result in scalability - run whatever test suite you want against any VM in parallel.

In regards to ostree upgrade/rollback I have this idea (not sure how it'll pan out):
if we go with "nested virt" approach, then we could setup a job (sanity periodic most likely) that would spawn multiple nested VMs and do some testing in parallel.
Like this mad idea: build 4.14.0 **base image** (as in qemu backing image, so new VMs can be spawned quickly) and do upgrades (rather by installing new RPMs than preparing whole new image)
```
 .0 ----
 |  \    \
.1   .2   .3
 |    |
.2   .3
 |
.3
```